File: 조성원/[LV2] 카펫.js
function solution(brown, yellow) {
  const total = brown + yellow;

  for (let h = 3; h <= Math.sqrt(total); h++) {
    if (total % h === 0) {
      const w = total / h;
      if (w >= h && w * 2 + (h - 2) * 2 === brown) return [w, h];
    }
  }
}

File: 김현경/[LV2] 카펫.py
def solution(brown, yellow):
    tot = brown + yellow
    
    for y in range(1, int(tot**0.5) + 1):
        if tot % y == 0:
            x = tot // y
            
            if (x-2) * (y-2) == yellow:
                return [x, y]